src/authenticator.rs

```diff
@@ -1087,6 +1080,17 @@ where
         reply?.retries
     }
 
+    fn _extension_get_key_for_pin(&mut self, password: &[u8]) -> Result<KeyId> {
+        let reply = try_syscall!(self.trussed.get_pin_key(
+            BACKEND_USER_PIN_ID,
+            Bytes::from_slice(password).map_err(|_| iso7816::Status::IncorrectDataParameter)?
+        ))
+        .map_err(|_| iso7816::Status::UnspecifiedNonpersistentExecutionError)?;
+        reply
+            .result
+            .ok_or(iso7816::Status::SecurityStatusNotSatisfied)
```

Comment:
This should be `Status::VerificationFailed` I think. The `get_pin_key` syscall performs the checking itself.

Reply:
Right, corrected